### ISSUE TITLE
ignore option in .babelrc is not recursive

### ISSUE THREAD

**Issue by gcedo** [babel/babel#4568]
This bug relates to #3789. The `ignore` option is no longer ignored, but the specified folders are not recursively ignored.
### Babel Configuration (.babelrc)

``` js
{
  "ignore": ["node_modules"]
}
```
### Expected Behavior

The `node_modules` directory and all of its files and subdirectories are ignored.
### Current Behavior

The build to fails as Babel will try to transpile files deep down in the `node_modules` hierarchy anyway:

```
69% 1206/1215 build modulesnode_modules/babel-plugin-react-transform/lib/index.js -> dist/node_modules/babel-plugin-react-transform/lib/index.js
```
### Your Environment

| software | version |
| --- | --- |
| Babel | 6.14.0 |
| node | 6.5.0 |
| npm | 3.10.3 |
| Operating System | Mac OSX 10.11.5 |

**Comment by bdentino** [comment]
Any plans to address this? It seems that the list of ignores specified in .babelrc is once again not being respected by babel-cli.

**Comment by aaronang** [comment]
@bdentino This should be fixed in #3644. Are you able to reproduce this?

**Comment by hettcher** [comment]
the same problem =(
http://take.ms/VEMYM

    "babel-preset-env": "^1.6.1",
    "babel-register": "^6.26.0",
    "babel-core": "^6.14.0",
    "gulp": "^4.0.0",
    "gulp-babel": "^7.0.0"
